corgi/core/models.py

```diff
@@ -1020,18 +1024,24 @@ def get_channels(self):
                     channels.append(descendant.obj.name)
         return list(set(channels))
 
-    def get_provides(self, include_dev=True):
-        """return all descendants with PROVIDES ComponentNode type"""
-        provides = set()
+    def get_provides_nodes(self, include_dev: bool = True) -> QuerySet[ComponentNode]:
+        """return a QuerySet of descendants with PROVIDES ComponentNode type"""
+        # Used in manifests. Returns whole objects to access their properties
         type_list = [ComponentNode.ComponentNodeType.PROVIDES]
         if include_dev:
             type_list.append(ComponentNode.ComponentNodeType.PROVIDES_DEV)
-        for cnode in self.cnodes.get_queryset():
-            if cnode.get_descendant_count() == 0:
-                continue
-            for descendant in cnode.get_descendants().filter(type__in=type_list):
-                provides.add(descendant.purl)
-        return list(provides)
+        return self.cnodes.get_queryset().get_descendants().filter(type__in=type_list)
```

Comment:
I'm gladly surprised that we don't have to iterate over the cnodes. I guess this must return the same results because you didn't have to alter the tests.

I'd be tempted to double check that a component with multiple cnodes (like a container with rpms and other types) still returns the same list of provides as before.

Reply:
I ran the below code which seems to still work:
```
from corgi.core.models import *
component_qs = Component.objects.get_queryset().iterator(chunk_size=500)

for component in component_qs:
     node_qs = component.cnodes.get_queryset()
     node_count = node_qs.count()

     if node_count < 2:
         continue

     descendant_count = node_qs.get_descendants().count()

     if descendant_count < 2:
         continue

     print("Descendants:", descendant_count)
     print("Nodes:", node_count)

     for node in node_qs:
         print(node.get_descendants().count())

     break
```
I got the following output, AKA calling get_descendants() on the whole queryset or on each individual node in the queryset gives the same result either way:
```
Descendants: 28
Nodes: 4
7
7
7
7
```